src/processors/BaseProcessor.h

```diff
@@ -186,6 +250,16 @@ class BaseProcessor : private JuceProcWrapper
      */
     auto& getSharedConvolutionMessageQueue() { return convolutionMessageQueue.get(); }
 
+    enum class BasicInputPort
+    {
+        AudioInput,
+    };
+
+    enum class BasicOutputPort
+    {
+        AudioOutput,
+    };
+
```

Comment:
It might make sense to add an empty `NullPort` enum class so that processors that have either no inputs (InputProcessor) or no outputs (Tuner, Oscilloscope, etc) can use it.